docs/content/dagster-cloud/developing-testing/dagster-cloud-yaml.mdx

```diff
@@ -0,0 +1,169 @@
+
+```shell
+quickstart_etl
+├── README.md
+├── quickstart_etl
+│   ├── __init__.py
+│   ├── assets
+├── quickstart_etl_tests
+├── dagster_cloud.yaml
+├── pyproject.toml
+├── setup.cfg
+└── setup.py
+```
+
+## Code Source
+
+The only **mandatory** section in the dagster_cloud.yaml file is setting up your code locations. If you have multiple code locations, you will have multiple code locations within a single dagster_cloud.yaml file. Dagster code can be sourced from a package, a python file, or a module.
+
+Parameter reference table:
+
+| Parameter                            | Description                                                                           | Data type            |
+| ------------------------------------ | ------------------------------------------------------------------------------------- | -------------------- |
+| `locations.location_name`            | The name of your code location that will appear in the Dagster UI code locations page | string               |
+| `locations.code_source.package_name` | The folder that contains your dagster code                                            | string (folder name) |
+| `locations.code_source.python_file`  | The python file that contains your dagster code                                       | string (.py file)    |
+| `locations.code_source.module_name`  | The module that contains your dagster code                                            | string (module name) |
+
+Single code location:
+
+```yaml
+#dagster_cloud.yaml
+
+locations:
+  - location_name: example_location
+    code_source:
+      package_name: my_dagster_project
+    build:
+      directory: ./
+      registry: 764506304434.dkr.ecr.us-west-2.amazonaws.com/branch-deployments-gh-action-test
+```
+
+Multiple code locations:
+
+```yaml
+#dagster_cloud.yaml
+
+locations:
+  - location_name: data-eng-pipeline
+    code_source:
+      package_name: hooli_data_eng
+  - location_name: basics
+    code_source:
+      python_file: definitions.py
+```
+
+## Configuring your working directory
+
+There might be some cases where you want to run your Dagster code from a different directory than the root of your code repository. You can specify the directory you want to run your Dagster code from in the dagster_cloud.yaml file using the `working_directory` option.
+
+Parameter reference table:
+
+| Parameter           | Description                                            | Data type     |
+| ------------------- | ------------------------------------------------------ | ------------- |
+| `working_directory` | The path of the directory that Dagster should run from | string (path) |
+
+```yaml
+#dagster_cloud.yaml
+
+locations:
+  - location_name: data-eng-pipeline
+    code_source:
+      package_name: hooli_data_eng
+    working_directory: ./hooli_data_eng
+```
+
+## Build: Docker Registry
+
```

Comment:
We should add a pre-amble here.

If you are using Dagster Cloud Hybrid with a Docker, Kubernetes, or ECR agent you will need to tell Dagster Cloud where the Docker image containing your code is stored. This is done through the`build` section of dagster_cloud.yaml.

Reply:
@gibsondan is this all true? I'm unclear as to what situations require `build` vs just including the `image` value. I think it has to do with whether you use our build_and_deploy action or just deploy... Any ideas on what language could make that distinction clear here?

Reply:
This section should just be called the "Build directory". You can use a `build: directory:` in serverless as well. It works for both docker and pex. Basically it specifies which directory contains your setup.py or requirements.txt file for this code location.

Reply:
We should say this is useful if your setup.py or requirements.txt is not at your project root but in a subdirectory. "This is common if you have multiple python packages within a single dagster project"